internal/shellgen/tmpl/flake_remove_nixpkgs.nix.tmpl

```diff
@@ -29,25 +37,59 @@
             {{- end }}
             {{- end }}
           ];
+
+          overlays = [
+            {{- range $flake.Packages }}
+            {{- if .PatchGlibc }}
+            (final: prev: {
+              {{.PackageAttributePath}} = prev.{{.PackageAttributePath}};
+            })
+            {{- end }}
+            {{- end }}
+          ];
```

Comment:
Can you comment on the tradeoffs between using an overlay versus a flake as done in https://github.com/savil/devbox-issue-1276? What led you down this path?

Reply:
So this overlay is actually wrong and could explain why virtualenvs are finding the wrong Python interpreter. I meant to set it to the patched version of the package.

From how I understand it, the overlay overrides the package for any other packages that depend on it. So if a python37Packages.package is installed, it should use the overridden python from the overlay. I'll fix this and see if it addresses the virtual env stuff.